chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/configuration/ChaosMonkeyConfiguration.java

```diff
@@ -142,24 +145,36 @@ public MemoryAssault memoryAssault() {
   public ChaosMonkeyRequestScope chaosMonkeyRequestScope(
       List<ChaosMonkeyRequestAssault> chaosMonkeyAssaults,
       List<ChaosMonkeyAssault> allAssaults,
-      ChaosToggles chaosToggles) {
+      ChaosToggles chaosToggles,
+      ChaosToggleNameMapper chaosToggleNameMapper) {
     return new ChaosMonkeyRequestScope(
-        settings(), chaosMonkeyAssaults, allAssaults, publisher(), chaosToggles);
+        settings(),
+        chaosMonkeyAssaults,
+        allAssaults,
+        publisher(),
+        chaosToggles,
+        chaosToggleNameMapper);
   }
 
   @Configuration
   @ConditionalOnClass(Unleash.class)
+  @ConditionalOnProperty(value = "chaos.monkey.toggle.unleash.enabled", matchIfMissing = true)
   static class UnleashChaosConfiguration {
```

Comment:
Please extract this class into a separate file so ChaosMonkeyConfiguration is kept as small as possible